### DISCUSSION TITLE
Question: Why MenuItems does not found on NavigationViewItem

### DISCUSSION THREAD

**Discussion by TheColdPot** [microsoft/microsoft-ui-xaml#7485] [Q&A]
(*I am using Visual Studio Chinese translated, the sentence below is translated which may be difference than Oringinal.*)
When I typed these code, editor tells me `Unexpected symbol "MenuItems"`
```xaml
<NavigationViewItem Content="Nav Item">
  <NavigationViewItem.MenuItems>
  </NavigationViewItem.MenuItems>
</NavigationViewItem>
```
Why and what happens?
I even changed my IDE from VS to JB, it's still not working
And also, this is not working
```xaml
<Page  ... xmlns:muxc="using:Windows.UI.Xaml.Controls" ... >
   ...
  <Grid>
    <muxc:NavigationViewItem Content="Nav Item">
      <muxc:NavigationViewItem.MenuItems>
      </muxc:NavigationViewItem.MenuItems>
    </muxc:NavigationViewItem>
  </Grid>
</Page>
```

**Comment by huoyaoyuan** [comment]
What version of WinUI are you using? It was introduced in very recent version.

**Comment by omikhailov** [comment]
There is no property "contents", try "Content" instead

**Comment by StephenLPeters** [comment]
@omikhailov is correct, contents is not a property on NavigationViewItem, the name of the property is **content**.

MenuItems was added to NavigationViewItem in Winui 2.4 (or maybe even 2.5...) make sure you have an updated winui2 package.

If both of those don't fix it, can you try adding a NavigationViewItem to the MenuItems. Its possible that an empty collection is not supported.

**Comment by TheColdPot** [comment]
> @omikhailov is correct, contents is not a property on NavigationViewItem, the name of the property is **content**.
> 
> MenuItems was added to NavigationViewItem in Winui 2.4 (or maybe even 2.5...) make sure you have an updated winui2 package.
> 
> If both of those don't fix it, can you try adding a NavigationViewItem to the MenuItems. Its possible that an empty collection is not supported.

- `NavigationViewItem.MenuItems` was add in WinUI 2.4

Sorry I have a wrong typing in issue, it's caused by myself, in my code it is `Content` instead `contents`
(*I fixed my issue body*)
This app was I created last week, so it should be last version of WinUI as expected (I'm using UWP)

**All normal `NavigationViewItem` is work as expected**, only `NavigationMenuItem.MenuItems` is wrong.

I can re-create my Project/Solution. I'm using Visual Studio **2019** and JB Rider **2021.1**. Will VS version cause this error? 
What should I do next?

**Comment by omikhailov** [comment]
You can try to change namespace to **Microsoft**.UI.Xaml.Controls. Windows.UI.Xaml.Controls contains platform controls, not WinUI.

**Comment by TheColdPot** [comment]
> You can try to change namespace to **Microsoft**.UI.Xaml.Controls. Windows.UI.Xaml.Controls contains platform controls, not WinUI.

"*Cannot resolve namespace `Microsoft.UI.Xaml.Controls`*", but `Microsoft.UI.Xaml` was exist.
An important thing is I'm using UWP instead pure WinUI

**Comment by marcelwgn** [comment]
You will need to add a reference to the [Microsoft.UI.Xaml NuGet package](https://www.nuget.org/packages/Microsoft.UI.Xaml) and follow the installation steps outlined in the packages readme.

**Comment by TheColdPot** [comment]
> You will need to add a reference to the [Microsoft.UI.Xaml NuGet package](https://www.nuget.org/packages/Microsoft.UI.Xaml) and follow the installation steps outlined in the packages readme.

Thanks, you resolved my problem! 

*If this issue is unimportant, is can be close now, or else it should be open for other developers that has same question*

**Comment by huoyaoyuan** [comment]
> What version of WinUI are you using?

So the awnser is effectively that you haven't install and aren't actually using WinUI. Or, you are using system UWP which can be also called WinUI 1.

**Comment by TheColdPot** [comment]
My fault. A package should be installed. If no further things, this issue can be closed.
